index.js

```diff
@@ -616,6 +616,9 @@ function speedTest(options) {
   nextServer();
 
   function gotServers(err, servers) {
+    if (!servers.hasOwnProperty("settings") || !servers.settings.hasOwnProperty("servers") || servers.settings.servers.length == 0) {
```

Comment:
If there's an error already, servers might be `undefined`. Would you mind fixing and simplifying this to `if (!err && (!servers.settings || !servers.settings.servers || !servers.settings.servers.length)) {` ?